### ISSUE TITLE
mip_improving_solution_save not saving on each improvement

### ISSUE THREAD

**Issue by Thell** [ERGO-Code/HiGHS#1857]
Hello!

Consider the following options being passed to HiGHS:

```
"mip_improving_solution_save=on",
"mip_improving_solution_file=highs_improved_solution_{max_cost}_c.sol",
```

I can see that they are being accepted because __a__ `highs_improved_solution` file was created and saved the first of the improved solutions.

That was ~6.5hrs ago. Since then over a dozen improvements have taken place yet none of them got saved. Much to my dismay because I just hit ctrl-c to copy something but my focus was on my terminal 🤦‍♂️ The real sad part was that I had realized this was taking place and the ctrl-c I used was to copy the options pasted above for this ticket. Anyhow, after getting down to a gap of 1.72% and 75.01% explored in those 6.5hrs I _really_ would like to have seen the last incumbent.

**Comment by jajhall** [comment]
How agonizing! I'll copy this to #1856

**Comment by Thell** [comment]
Just an FYI, this isn't via Highspy, this is via Pulp using HiGHS_CMD which writes out the LP and then calls the HiGHS executable. I admit I haven't tested calling the executable via the command line with an options file to test it yet. But I guess if _something_ wonky might be going on that would tell me. I'll test it out tomorrow and let you know.

**Comment by Thell** [comment]
Well, color me embarrassed. 😊 

I went to test out the improved file writing from the terminal and noted the same issue but then noticed that the file timestamp updated... it is _appending_, so when I looked at the objective in the first line it looked like it didn't update. Grepping the file showed all of the improved solutions. Grepping the improved solution file being used via Pulp -> HiGHS_CMD shows it is updating (appending).

Mark it up to ignorance.
